epoxy-paging/src/main/java/com/airbnb/epoxy/paging/PagedListModelCache.kt

```diff
@@ -31,130 +32,222 @@ import java.util.concurrent.Executor
  * A PagedList stream wrapper that caches models built for each item. It tracks changes in paged lists and caches
  * models for each item when they are invalidated to avoid rebuilding models for the whole list when PagedList is
  * updated.
+ *
+ * The PagedList submitted to this cache must be kept in sync with the model cache. To do this,
+ * the executor of the PagedList differ is set to the same thread as the model building handler.
+ * However, change notifications from the PageList happen on that list's notify executor which is
+ * out of our control, and we require the user to configure that properly, or an error is thrown.
+ *
+ * There are two special cases:
+ *
+ * 1. The first time models are built happens synchronously for immediate UI. In this case we don't
+ * use the model cache (to avoid data synchronization issues), but attempt to fill the cache with
+ * the models later.
+ *
+ * 2. When a list is submitted it can trigger update callbacks synchronously. Since we don't control
```

Comment:
this looks like a paging bug. we should not be notifying on another thread. though paging receives just an executor so probably paging is unable to check that :/.